controllers/order.js

```diff
@@ -0,0 +1,139 @@
+//const Order = require("../models/Order");
+const Order = require("../models/Order");
+const { validationResult } = require("express-validator");
+
+
+exports.createOrder = async (req, res, next) => {
+  try {
+    const {
+      user,
+      items,
+      cartDetails,
+      product,
+      paymentStatus,
+      paymentType,
+      orderStatus,
+      type,
+    } = req.body;
+    const order = await Order.create({
+      user, 
+      items,
+      cartDetails,
+      product,
+      paymentStatus,
+      paymentType,
+      orderStatus,
+      type,
+    });
+
+    return res.status(200).json({
+      success: true,
+      message: "Order Placed successfully.",
+      data: order,
+    });
+  } catch (error) {
+    console.log(error);
+    return res.status(500).json({
+      success: false,
+      message:
+        "We are having some error while completing your request. Please try again after some time.",
+      error: error,
+    });
+  }
+};
+
+exports.getOne = async (req, res) => {
+  try {
+    const order = await Order.findById(req.params.id)
+      .select("address  paymentStatus paymentType orderStatus")
+     
+      .populate("items.product")
+      .populate("cartDetails");
+    if (!order) {
+      return res.status(400).json({ error: " Order not found...... " });
+    }
+    res.status(200).json({
+      status: true,
+      Order: order,
+    });
+  } catch (err) {
+    res.status(500).json(err);
+  }
+};
+
+exports.updateOrder = async (req, res) => {
+  try {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      res.status(422).json({ errors: errors.array() });
+      return;
+    }
+    const {
+      user,
+     
+      items,
+      cartDetails,
+      product,
+      paymentStatus,
+      paymentType,
+      orderStatus,
+      type,
+    } = req.body;
+    const updatedOrder = await Order.findByIdAndUpdate(req.params.id, {
+      $set: {
+        user,
+        
+        items,
+        cartDetails,
+        product,
+        paymentStatus,
+        paymentType,
+        orderStatus,
+        type,
+      },
+    });
+    res.status(200).json({
+      message: "Order updated successfully",
+      data: updatedOrder,
+    });
+  } catch (err) {
+    res.status(500).json(err);
+  }
+};
+
+exports.allOrders = async (req, res) => {
+  try {
+    const { limit = 10, page = 1 } = req.query;
+    const sort = {};
+    let filter = {};
+
+    if (req.query.sortBy) {
+      const str = req.query.sortBy.split(":");
+      sort[str[0]] = str[1] === "desc" ? -1 : 1;
+    }
+    const orderList = await Order.find(filter)
+      .populate("cartDetails")
+      .populate("items.product")
+      .sort(sort)
+      .limit(limit)
+      .skip(limit * (page - 1));
+    const totalItems = await Order.countDocuments(filter);
+    if (!orderList) {

```

Comment:
will this work?

because query will return empty array and that will consider as true